### ISSUE TITLE
[PPML] PPMLContext doesn't read and decrypt ciphertext correctly

### ISSUE THREAD

**Issue by SEULiyao** [intel-analytics/BigDL#7400]
When the input file is encrypted data, PPMLContext doesn't read and decrypt correctly. It outputs garbled code during SQL operation, which means that the ciphertext has not been decrypted correctly.
```
Exception in thread "main" org.apache.spark.sql.AnalysisException: Cannot resolve column name "name" among (�cڇ4~2�xA��4�, ��`���\ji�oe}\8�q���, ȷD:2�M�蠤��O��I�&�mES��n��ڼ[)
```

It is found that the [`EncryptedDataFrameReader.csv(path: String)`](https://github.com/intel-analytics/BigDL/blob/main/scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptedDataFrameReader.scala#L43) method does not implement the reading of csv ciphertext.
```scala
  def csv(path: String): DataFrame = {
    sparkSession.read.options(extraOptions).csv(path)
  }
```

But `EncryptedDataFrameReader.parquet` method has different treatment for the case of plaintext and that of ciphertext respectively:
```scala
  def parquet(path: String): DataFrame = {
    if (encryptMode != PLAIN_TEXT) {
      EncryptedDataFrameReader.setParquetKey(sparkSession, dataKeyPlainText)
    }
    sparkSession.read.parquet(path)
  }
```
And [`EncryptedDataFrameWriter`](https://github.com/intel-analytics/BigDL/blob/main/scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptedDataFrameWriter.scala#L64) also deals with different cases separately:
```scala
  def csv(path: String): Unit = {
    encryptMode match {
      case PLAIN_TEXT =>
        df.write.options(extraOptions).mode(mode).csv(path)
      case AES_CBC_PKCS5PADDING =>
        sparkSession.sparkContext.hadoopConfiguration
          .set("hadoop.io.compression.codecs",
          "com.intel.analytics.bigdl.ppml.crypto.CryptoCodec")
        df.write
          .option("compression", "com.intel.analytics.bigdl.ppml.crypto.CryptoCodec")
          .options(extraOptions).mode(mode).csv(path)
      case _ =>
        Log4Error.invalidOperationError(false, "unknown EncryptMode " + CryptoMode.toString)
    }
  }
```

As a result, `sc.read(AES_CBC_PKCS5PADDING)` presented in [doc](https://github.com/intel-analytics/BigDL/blob/main/ppml/docs/examples.md#run-trusted-simplequery) is not correctly implemented: in PPML E2E applications, data uploaded to cloud/cluster has to be plaintext actually, which is an unsafe operation and  does not fit our scenario design.

```bash
# in bigdl-ppml-submit command
...
--inputEncryptModeValue AES/CBC/PKCS5Padding \ # does not work and is plain text actually
...
```

**Comment by Uxito-Ada** [comment]
Encrypted input is judged by whether its suffix is ["cbc"](https://github.com/intel-analytics/BigDL/blob/main/scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/CryptoCodec.scala#L94) while you need to register the codec to your hadoop conf in read APIs.

**Comment by Uxito-Ada** [comment]
Two points are found about this issue:

1. CryptoCodec has not been registered correctly in PPMLContext. There is a difference between DataFrameReader and DataFrameWriter that the latter provides a "compression" option interface to register codec for each writer, while all DataFrameReaders share a global codec which is immutable after SparkSession is initialized, and **this cause codec set for reader invalid in initPPMLContext**.

In conclusion, we need to **remove initPPMLContext(SparkSession) API for correct codec registration.**

2. CryptoCodec for EncryptedDataFrameReader can cause key conflict and decryption failure: this is because all reader instances shares the same key filed in global hadoop conf (since CryptoCodec does not accepts any input parameter), and it is difficult to express non-peer such a crypto computation in RDD programming model.
 
We can fix it by **wrap a df.cache.count action in backend reader.csv method** which splits read operations into different stages, while this also degrades to serial execution.


we are WIP to fix this issue in next version of PPML Multi-Party feature.

**Comment by Uxito-Ada** [comment]
fix links to #7445
